### PR TITLE
Added analytics to file transfer and job submission

### PR DESCRIPTION
**Pull request by adamfitzgibbon**
Added events to track file transfer and job submission. This should actually be tested because I don't have access to the google analytics.

### CODE REVIEW

**Comment by adamfitzgibbon** [comment]
Yeah I saw that, I'm not sure why it does that. I only wanted to fix the tabbing, but sometimes Atom just does that weird whitespace change when I save a new file like that.

**Comment by djw8605** [comment]
Did you ever figure out the issue with getting "Transfer Failed" even though the transfer was successful?

**Comment by adamfitzgibbon** [comment]
I haven't but I also haven't had a chance to ask Jerrod about it.

**Comment by adamfitzgibbon** [comment]
@robogen Can you take a look at the changes in ConnectionService? I think I put in one too many fail event calls because when I successfully download a file both a success event and a fail event are thrown, but I'm not sure which of the fail events is not supposed to be there. I'm guessing there's somewhere that I think is supposed to only execute if there is an error, but it executes no matter what.

**Comment by djw8605** [comment]
Does the analytics send an event for each and every file transferred?  Or an aggregate for all transfers?

**Comment by adamfitzgibbon** [comment]
I went straight to the uploadFile and downloadFile methods in ConnectionService, so I'm assuming that would be every file transferred unless those methods allow for groupings like that.

**Comment by adamfitzgibbon** [comment]
I just merged in Jerrod's changes, so I'm going to do a file upload and download to see if it successfully sends the events now.

**Comment by djw8605** [comment]
Oh, could it be an aggregate for each upload / download task?  I'm worried that it will be too many events if we do it for each file.

**Comment by adamfitzgibbon** [comment]
Wouldn't that somewhat compromise the data though? If we group them together then we'll only see entire batches failing if one of them fails. Perhaps that is all we need to know though?

**Comment by djw8605** [comment]
Yeah.  Though if 1 transfer fails, are they all marked as failed?

**Comment by adamfitzgibbon** [comment]
I'm not sure exactly hows errors are being thrown, but I had guessed that it was just calling the respective functions multiple times for each file if a directory was specified. So if that's correct than I would assume that errors would be handled separately for each file, but again, I'm not sure on that.

**Comment by djw8605** [comment]
@robogen can answer for sure.  But I think there's an `async.queue` involved.

**Comment by robogen** [comment]
For the actual transferring I'm using `async.cargo` but yes, it works on an aggregate sense. There's a 'persistent' err variable used through the function, persistent mostly in name though. The value gets forwarded through the application via callbacks. With normal behaviour, err never has anything other than a null value. If however during the execution an issue does occur than err is initialized with the error code, results, stack trace, etc. Because of that, when err is non-null it breaks the execution path and produces an error log.

tldr; Yeah, it'll produce an aggregate error, though messages relating to specific files will get logged out.

**Comment by adamfitzgibbon** [comment]
Everything should be working now, and it should only send an event on a successful batch, not per file.
